src/api/lint.ts

```diff
@@ -1,84 +1,38 @@
-
-const getExtensions = (
-  typescript?: boolean,
-  javascript?: boolean,
-  includeJsx?: boolean,
-): string[] => {
-  let extensions: string[] = [];
-  if (typescript) {
-    extensions.push('.ts');
-  }
-  if (javascript) {
-    extensions.push('.js');
-  }
-  if (includeJsx) {
-    extensions = extensions.concat(getJsxEtensions(extensions));
-  }
-  return extensions;
-};
+import { lintLogic } from './lint-logic';
+import { lintStyle } from './lint-style';
+import { LintOptions, LintResult } from './types';
+import { getConfig } from './config';
 
 const hasAnyLintingSet = ({
   javascript,
   typescript,
   sass,
 }: LintOptions): boolean => !!javascript || !!typescript || !!sass;
 
-const setDefaultOptions = (options: LintOptions): LintOptions => ({
+const parseOptions = (options: LintOptions): LintOptions => ({
   ...options,
   pattern: options.pattern || './',
   javascript: !hasAnyLintingSet(options) || options.javascript,
   typescript: !hasAnyLintingSet(options) || options.typescript,
+  sass: !hasAnyLintingSet(options) || options.sass,
 });
 
-export const lint = (options: LintOptions): LintReport => {
-  const result: LintReport = {};
-  const {
-    config,
-    pattern,
-    typescript,
-    javascript,
-    includeJsx,
-    sass,
-    fix,
-    spec,
-  } = setDefaultOptions(options);
-
-  const configFile = getConfigFile(config);
-
-  if (typescript || javascript) {
-    const extensions = getExtensions(typescript, javascript, includeJsx);
-    const testExtensions = getTestExtensions(extensions);
-    const patterns = (pattern && pattern.split(',').filter(p => !!p)) || [];
-    const baseConfig: Eslint.Options['baseConfig'] = {
-      ...(includeJsx ? { parserOptions: { ecmaFeatures: { jsx: true } } } : {}),
-      // eslint-disable-next-line global-require,import/no-dynamic-require
-      ...require(configFile),
-    };
+export const lint = (options: LintOptions): Promise<LintResult> => {
+  const parsedOptions = parseOptions(options);
+  const config = getConfig(options.config);
+  const linterPromises: Promise<LintResult['logic'] | LintResult['style']>[] = [];
 
-    result.logic = (new Eslint({
-      useEslintrc: false,
-      baseConfig,
-      fix,
-      extensions: !spec ? extensions : testExtensions,
-      ignorePattern: !spec ? testExtensions.map(ext => `*${ext}`) : undefined,
-    })).executeOnFiles(patterns);
-
-    Eslint.outputFixes(result.logic);
+  if (parsedOptions.sass) {
+    linterPromises.push(lintStyle(config, parsedOptions));
   }
 
-  if (sass) {
-    // eslint-disable-next-line no-console
-    console.log('SCSS Linting: Not implemented yet…');
-
-    result.style = {};
+  if (parsedOptions.typescript || parsedOptions.javascript) {
+    linterPromises.push(lintLogic(config, parsedOptions));
   }
 
-  return result;
+  return new Promise((resolve): Promise<LintResult | void> => Promise.all(linterPromises)
```

Comment:
Could you explain me why are you using a new promise to resolve another promise?

Reply:
Old leftover code. Fixed.